compiler/src/iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp

```diff
@@ -46,50 +46,76 @@ VulkanSPIRVTargetOptions getVulkanSPIRVTargetOptionsFromFlags() {
   // llvm::cl::OptionCategory halVulkanSPIRVOptionsCategory(
   //     "IREE Vulkan/SPIR-V backend options");
 
-  static llvm::cl::opt<std::string> clVulkanTargetTriple(
+  static llvm::cl::list<std::string> clVulkanTargetTriples{
       "iree-vulkan-target-triple",
       llvm::cl::desc(
           "Vulkan target triple controlling the SPIR-V environment."),
-      llvm::cl::init("unknown-unknown-unknown"));
+  };
 
-  static llvm::cl::opt<std::string> clVulkanTargetEnv(
+  static llvm::cl::list<std::string> clVulkanTargetEnvs{
       "iree-vulkan-target-env",
       llvm::cl::desc(
           "Vulkan target environment as #vk.target_env attribute assembly."),
-      llvm::cl::init(""));
+  };
 
   static llvm::cl::opt<bool> clVulkanIndirectBindings(
       "iree-vulkan-experimental-indirect-bindings",
       llvm::cl::desc("Force indirect bindings for all generated dispatches."),
       llvm::cl::init(false));
 
   VulkanSPIRVTargetOptions targetOptions;
-  targetOptions.targetEnv = clVulkanTargetEnv;
-  targetOptions.targetTriple = clVulkanTargetTriple;
+
+  int tripleCount = clVulkanTargetTriples.getNumOccurrences();
+  int envCount = clVulkanTargetEnvs.getNumOccurrences();
+  int tripleIdx = 0;
+  int envIdx = 0;
+
+  // Use the relative positions in the argument list to get the flat list of
```

Comment:
Maybe something like "Get a flat list of target triples and environments following the original order specified via the command line"?